app/agents/voice/automatic/tools/juspay/analytics.py

```
@@ -155,6 +159,142 @@ async def _make_genius_api_request(
+            )
+        start_time_utc = start_time_ist.astimezone(utc)
+
+        if end_time_str:
+            end_time_ist = ist.localize(
+                datetime.strptime(end_time_str, "%Y-%m-%d %H:%M:%S")
+            )
+        else:
+            end_time_ist = datetime.now(ist)
+        end_time_utc = end_time_ist.astimezone(utc)
+
+    except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)